Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -732,49 +731,71 @@ static void ArrayBindTest(string connstr, string tableName, int size)
                     //cmd.ExecuteNonQuery();
 
                 }
-                System.Threading.Thread.Sleep(2000); // wait for 2 seconds, in case other test still running.
+                //System.Threading.Thread.Sleep(2000); // wait for 2 seconds, in case other test still running.
                 conn.Close();
             }
         }
 
         [Test]
         public void TestPutArrayBindAsyncMultiThreading()
         {
-            var connStr = ConnectionString + "";
-            Thread t1 = new Thread(() => ThreadProcess1(ConnectionString));
-            Thread t2 = new Thread(() => ThreadProcess2(ConnectionString));
+            var t1TableName = TableName + 1;
+            var t2TableName = TableName + 2;
+            CreateOrReplaceTable(t1TableName, new []
+            {
+                "cola INTEGER",
+                "colb STRING",
+                "colc DATE",
+                "cold TIME",
+                "cole TIMESTAMP_NTZ",
+                "colf TIMESTAMP_TZ"
+            });
+            CreateOrReplaceTable(t2TableName, new []
+            {
+                "cola INTEGER",
+                "colb STRING",
+                "colc DATE",
+                "cold TIME",
+                "cole TIMESTAMP_NTZ",
+                "colf TIMESTAMP_TZ"
+            });
+            Thread t1 = new Thread(() => ThreadProcess1(ConnectionString, t1TableName));
+            Thread t2 = new Thread(() => ThreadProcess2(ConnectionString, t2TableName));
             //Thread t3 = new Thread(() => ThreadProcess3(ConnectionString));
             //Thread t4 = new Thread(() => ThreadProcess4(ConnectionString));
 
             t1.Start();
             t2.Start();
             //t3.Start();
             //t4.Start();
+            t1.Join();
```

Comment:
why introduced those t1.Join and t2.Join new lines?

Reply:
We have to wait for the test cases to complete. Without joining the threads the method finishes before the threads complete their tests.